packages/@lwc/engine-core/src/framework/decorators/track.ts

```diff
@@ -13,27 +13,32 @@ import { LightningElement } from '../base-lightning-element';
 import { isUpdatingTemplate, getVMBeingRendered } from '../template';
 import { updateComponentValue } from '../update-component-value';
 
+import { ClassFieldDecorator } from './types';
+import { validateFieldDecoratedWithTrack } from './register';
+
 /**
  * @track decorator function to mark field value as reactive in
  * LWC Components. This function can also be invoked directly
  * with any value to obtain the trackable version of the value.
  */
-export default function track(
-    target: any,
-    propertyKey: string,
-    descriptor: PropertyDescriptor
-): any;
-export default function track(target: any): any {
-    if (arguments.length === 1) {
-        return getReactiveProxy(target);
-    }
-    if (process.env.NODE_ENV !== 'production') {
-        assert.fail(
-            `@track decorator can only be used with one argument to return a trackable object, or as a decorator function.`
-        );
-    }
-    throw new Error();
-}
+const track: ClassFieldDecorator = (value, context) => {
+    if (context.kind == 'field') {
+        const { name } = context;
+        return function (this: LightningElement, initialValue) {
+            if (process.env.NODE_ENV !== 'production') {
+                const vm = getAssociatedVM(this);
+                const descriptor = getOwnPropertyDescriptor(vm.def.ctor, name);
+                if (descriptor) {
+                    validateFieldDecoratedWithTrack(vm.def.ctor, name as string, descriptor);
+                }
+            }
+            defineProperty(this, name, internalTrackDecorator(name as string));
```

Comment:
I'm sure babel can do whatever babel does... but will this work with built-in implementation? I don't believe it does, at least last time I checked, there were certain restrictions on changing the `kind` during initialization from `field` to accessor.